pkg/maps/ctmap/ctmap.go

```diff
@@ -516,7 +516,12 @@ func DeleteIfUpgradeNeeded(e CtEndpoint) {
 			scopedLog.WithError(err).Debug("Couldn't open CT map for upgrade")
 			continue
 		}
-		if oldMap.CheckAndUpgrade(&newMap.Map.MapInfo) {
+		checkAndUpgrade, err := oldMap.CheckAndUpgrade(&newMap.Map.MapInfo)
+		if err != nil {
+			scopedLog.WithError(err).Fatal("Couldn't perform BPF map checks for CT map upgrade")
```

Comment:
Due to [1], depending on whether the ctmap is the first map to run or not plus if the probing is somehow unsuccessful only from the `GetMapType()` path (but not in the initial invocation of `NewProbeManager()`), this `Fatal()` may actually be avoided even if the map types were not successfully probed.